### DISCUSSION TITLE
[ Feature ] 关于实现工程代码内的 SQL 审核的讨论

### DISCUSSION THREAD

**Discussion by nowindnowave** [hhyo/Archery#863] [Ideas]
### 问题描述
想咨询个问题，有没有工具可以检测java提交代码中的SQL语句？目前只看到慢查询和SQL优化，有没有从源头检测的工具？

**Comment by hhyo** [comment]
前审是很好的想法，可以参考#3

目前业界也有一些成熟的方案可以借鉴

**Comment by nowindnowave** [comment]
非常感谢

**Comment by nowindnowave** [comment]
我看最新回复是2019年1月
完成XML解析成SQL：https://github.com/hhyo/mybatis-mapper2sql
后续扩展下系统内SOAR的方法就可以实现自动审核

现在已经扩展soar实现自动审核xml了吗？
